lib/FIR/afforestation.cc

```diff
@@ -988,39 +209,70 @@ const parser::Format *FindReadWriteFormat(
 }
 
 static Expression AlwaysTrueExpression() {
-  using T = evaluate::Type<evaluate::TypeCategory::Logical, 1>;
-  return {evaluate::AsGenericExpr(evaluate::Constant<T>{true})};
+  using A = evaluate::Type<evaluate::TypeCategory::Logical, 1>;
+  return {evaluate::AsGenericExpr(evaluate::Constant<A>{true})};
 }
 
 // create an integer constant as an expression
 static Expression CreateConstant(int64_t value) {
-  using T = evaluate::SubscriptInteger;
-  return {evaluate::AsGenericExpr(evaluate::Constant<T>{value})};
+  using A = evaluate::SubscriptInteger;
+  return {evaluate::AsGenericExpr(evaluate::Constant<A>{value})};
 }
 
 static void CreateSwitchHelper(FIRBuilder *builder, Value condition,
-    BasicBlock *defaultCase, const SwitchStmt::ValueSuccPairListType &rest) {
-  builder->CreateSwitch(condition, defaultCase, rest);
+    const SwitchStmt::ValueSuccPairListType &rest) {
+  builder->CreateSwitch(condition, rest);
 }
 static void CreateSwitchCaseHelper(FIRBuilder *builder, Value condition,
-    BasicBlock *defaultCase,
     const SwitchCaseStmt::ValueSuccPairListType &rest) {
-  builder->CreateSwitchCase(condition, defaultCase, rest);
+  builder->CreateSwitchCase(condition, rest);
 }
 static void CreateSwitchRankHelper(FIRBuilder *builder, Value condition,
-    BasicBlock *defaultCase,
     const SwitchRankStmt::ValueSuccPairListType &rest) {
-  builder->CreateSwitchRank(condition, defaultCase, rest);
+  builder->CreateSwitchRank(condition, rest);
 }
 static void CreateSwitchTypeHelper(FIRBuilder *builder, Value condition,
-    BasicBlock *defaultCase,
     const SwitchTypeStmt::ValueSuccPairListType &rest) {
-  builder->CreateSwitchType(condition, defaultCase, rest);
+  builder->CreateSwitchType(condition, rest);
+}
+
+static Expression getApplyExpr(Statement *s) {
+  return GetApplyExpr(s)->expression();
+}
+static Expression getLocalVariable(Statement *s) {
+  return GetLocal(s)->variable();
 }
 
+// create a new temporary name (as heap garbage)
+static parser::CharBlock NewTemporaryName() {
+  constexpr int SizeMagicValue{32};
+  static int counter;
+  char cache[SizeMagicValue];
+  int bytesWritten{snprintf(cache, SizeMagicValue, ".t%d", counter++)};
+  CHECK(bytesWritten < SizeMagicValue);
+  auto len{strlen(cache)};
+  char *name{new char[len]};  // XXX: add these to a pool?
+  memcpy(name, cache, len);
+  return {name, name + len};
+}
+
+static TypeRep GetDefaultIntegerType() {
+  return {semantics::NumericTypeSpec{evaluate::SubscriptInteger::category,
+      AsExpr(evaluate::Constant<evaluate::SubscriptInteger>{
+          evaluate::SubscriptInteger::kind})}};
```

Comment:
In surfing around through other code, I found this `SubscriptInteger` being used.  I don't really know if it is "best" way to do this.  But perhaps the name of the function could be improved as well.  I went surfing for a "default", but in the context of how this is *actually* being used, at least one of the cases really is computing a new expression for accessing into array objects.  So that, I'm guessing, should likely be somehow tied to other subscript type manipulations.

Reply:
Ok, it makes sens to produce SubscriptInteger if you are using them for subscripts, maybe the function could just be renamed `GetSubscriptIntegerType` to avoid any confusion with the default integer that may or may not be a subscript integer.

Reply:
done